tests/lib/prepare-restore.sh

```diff
@@ -510,8 +512,10 @@ prepare_suite_each() {
 
     # save the job which is going to be executed in the system
     echo -n "$SPREAD_JOB " >> "$RUNTIME_STATE_PATH/runs"
-    # shellcheck source=tests/lib/reset.sh
-    "$TESTSLIB"/reset.sh --reuse-core
+    if [[ "$variant" = full ]]; then
```

Comment:
This doesn't need the [[ ]] syntax.
```suggestion
    if [ "$variant" = full ]; then
```